src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -308,6 +312,10 @@ size_t ArchiveBuilder::estimate_archive_size() {
   size_t dictionary_est = SystemDictionaryShared::estimate_size_for_archive();
   _estimated_hashtable_bytes = symbol_table_est + dictionary_est;
 
+  if (CDSConfig::is_dumping_aot_linked_classes()) {
+    _estimated_hashtable_bytes += _klasses->length() * 16 * sizeof(Klass*);
```

Comment:
Why 16?

Reply:
Doing the estimate is actually difficult here (and also pointless). I've filed https://bugs.openjdk.org/browse/JDK-8340416 to remove the estimation altogether. For the time being, I change the estimate to 20MB which will be more than enough.